src/x/cost/enforcer.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cost
+
+import (
+	"fmt"
+
+	"github.com/uber-go/tally"
+)
+
+const (
+	defaultCostExceededErrorFmt = "%s exceeds limit of %s"
+	customCostExceededErrorFmt  = "%s exceeds limit of %s: %s"
+)
+
+var (
+	noopManager = NewStaticLimitManager(
+		NewLimitManagerOptions().
+			SetDefaultLimit(Limit{
+				Threshold: maxCost,
+				Enabled:   false,
+			},
+			),
+	)
+	noopEnforcer = NewEnforcer(noopManager, NewNoopTracker(), nil)
+)
+
+// Report is a report on the cost limits of an enforcer.
+type Report struct {
+	Cost
+	Error error
+}
+
+// Enforcer instances enforce cost limits for operations.
+type Enforcer interface {
```

Comment:
this should also go in types.go and be tied into the documentation

Reply:
Sure I can move it. What do you mean tied into the documentation? `godocs` should already pick this up, no?